services/dynamic-sidecar/src/simcore_service_dynamic_sidecar/modules/attribute_monitor/_logging_event_handler.py

```diff
@@ -0,0 +1,226 @@
+from servicelib.logging_utils import log_context
+from watchdog.events import FileSystemEvent
+
+from ._watchdog_extensions import ExtendedInotifyObserver, SafeFileSystemEventHandler
+
+_HEART_BEAT_MARK: Final = 1
+
+logger = logging.getLogger(__name__)
+
+
+class _LoggingEventHandler(SafeFileSystemEventHandler):
+    # NOTE: runs in the created process
+
+    def event_handler(self, event: FileSystemEvent) -> None:
+        file_path = Path(event.src_path)
+        file_stat = file_path.stat()
+        logger.info(
+            "Attribute change to: '%s': permissions=%s uid=%s gid=%s size=%s\nFile stat: %s",
+            file_path,
+            stat.filemode(file_stat.st_mode),
+            file_stat.st_uid,
+            file_stat.st_gid,
+            ByteSize(file_stat.st_size).human_readable(),
+            file_stat,
+        )
+
+
+class _LoggingEventHandlerProcess:
+    def __init__(
+        self,
+        path_to_observe: Path,
+        health_check_queue: AioQueue,
+        heart_beat_interval_s: PositiveFloat,
+    ) -> None:
+        # NOTE: runs in asyncio thread
+
+        self.path_to_observe: Path = path_to_observe
+        self.health_check_queue: AioQueue = health_check_queue
+        self.heart_beat_interval_s: PositiveFloat = heart_beat_interval_s
+
+        # This is accessible from the creating process and from
+        # the process itself and is used to stop the process.
+        self._stop_queue: AioQueue = aioprocessing.AioQueue()
+
+        self._file_system_event_handler: Optional[_LoggingEventHandler] = None
+        self._process: Optional[AioProcess] = None
+
+    def start_process(self) -> None:
+        # NOTE: runs in asyncio thread
+
+        with log_context(
+            logger,
+            logging.DEBUG,
+            f"{_LoggingEventHandlerProcess.__name__} start_process",
+        ):
+            self._process = aioprocessing.AioProcess(
+                target=self._process_worker, daemon=True
+            )
+            self._process.start()
+
+    def _stop_process(self) -> None:
+        # NOTE: runs in asyncio thread
+
+        with log_context(
+            logger,
+            logging.DEBUG,
+            f"{_LoggingEventHandlerProcess.__name__} stop_process",
+        ):
+            self._stop_queue.put(None)
+
+            if self._process:
+                # force stop the process
+                self._process.kill()
+                self._process.join()
+                self._process = None
+
+            # cleanup whatever remains
+            self._file_system_event_handler = None
+
+    def shutdown(self) -> None:
+        # NOTE: runs in asyncio thread
+
+        with log_context(
+            logger, logging.DEBUG, f"{_LoggingEventHandlerProcess.__name__} shutdown"
+        ):
+            self._stop_process()
+
+            # signal queue observers to finish
+            self.health_check_queue.put(None)
+
+    def _process_worker(self) -> None:
+        # NOTE: runs in the created process
```

Comment:
minor: instead of all these notes, maybe having a prefix in the method would be clearer no? Or since almost everything runs in asyncio thread, maybe have a clearer name for the methods that do not run in asyncio thread...